docs/plugin_developers_guide/09_packaging.rst

```diff
@@ -0,0 +1,256 @@
+.. Documentation of package for a Plugin
+
+Packaging Documentation
+=======================
+
+Below are the prerequisite files which we need them in Plugin repository itself.
+
+Common files
+------------
+
+- Description
+   Summary of a plugin. Also make sure description of plugin must be in a single line as we do not support multiline yet.
+
+- Package
+   This is the main file where we define set of variables.
+
+   - plugin_name
+      Name of the Plugin.
+   - plugin_type
+      Type of the plugin.
+   - plugin_install_dirname
+      Installed directory name.
+   - plugin_package_name (OPTIONAL)
+      Name of the Package and also fullname is required. If not given or variable is not set then the package name should be same as plugin name.
+   - requirements
+      Runtime Architecture specific packages list in comma separated without any space.
+
+   .. note::
+      For C-based plugins if a plugin requires some additional libraries to install with then set additional_libs variable inside Package file. And the value must be with following contract:
+
+      additional_libs="DIRECTORY_NAME:FILE_NAME" - in case of single
+      additional_libs="DIRECTORY_NAME:FILE_NAME1,DIRECTORY_NAME:FILE_NAME2" - in case of multiple use comma separated with both directory & file name
+
+
+- service_notification.version
+   This file only requires if plugin is notification rule or delivery based.
```

Comment:
This file is only required if the plugin is a notification rule or notification delivery plugin. It contains the minimum version of the notification service which the plugin requires.